packages/ui/src/stories/Button/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+/* import './button.css'; */
+import ButtonStyles from "./button.style";
+
+export interface ButtonProps {
+  type: string;
+  id: string;
+  htmlType: string;
+  onClick: any;
+  disabled: boolean | undefined;
+  style: Object;
+  className: string;
+  children: any;
+}
+
+export function Button({
+  type,
+  id,
+  htmlType,
+  onClick,
+  children,
+  disabled,
+  style,
+  className,
+}: ButtonProps) {
+  return (
+    <ButtonStyles>
+      {/* eslint-disable-next-line */}
```

Comment:
If you remove this you will see typescript gives you a very descriptive message on how to annotate the `htmlType`. Please check it out and when this is solved please remove this line. Remember we want to avoid disabling eslint in the actual code 99.9 percent of the time.

https://github.com/chakra-ui/chakra-ui/blob/4ce9e3c0f6/packages/button/src/button.tsx

Check this library and you will also find how to annotate the functions ;)

Reply:
Already removed the eslint line, same mistake as with the other one.

Regarding the htmlType annotations. I have been reading and I already understand what it means, and I also understand this annotation:

> The expected type comes from property 'type' which is declared here on type 'DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>'

From what I got React exports an interface HTML Elements and this interface has its own types:

` interface ButtonHTMLAttributes<T> extends HTMLAttributes<T> {
        autoFocus?: boolean;
        disabled?: boolean;
        form?: string;
        formAction?: string;
        formEncType?: string;
        formMethod?: string;
        formNoValidate?: boolean;
        formTarget?: string;
        name?: string;
        type?: 'submit' | 'reset' | 'button';
        value?: string | ReadonlyArray<string> | number;
    }`
    
  That is why Typescript complains when I use a type different than  'submit' | 'reset' | 'button' | undefined. Is that right?
  
  I am not sure how to solve it though, because we don't want to use those types, we want to use `primary`, `secondary`, etc, is that right?
  
   For that reason I am not sure what to do, I suppose we shouldn't change React's interface.

Reply:
More on annotation.
 I decided to follow the advice of  [this](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example) article, it is a cheat-sheet from Netlify, which I believe is a trust able company, is it? 
 Because of that I used the type `(event: React.MouseEvent<HTMLButtonElement>) => void` for the `onClick` function.
 
 For the children I used `React.ReactNode` because according to them is the best one. Nevertheless, it is a bit weird to me that children from a button can be virtually anything. Is it right to use `React.ReactNode`? I suppose we want to include cases when we use tags like `<span>` or `<icon>`

Reply:
you are 100 percent right on your research! Good work :)